dateparser/date.py

```diff
@@ -304,6 +314,11 @@ def get_date_data(self, date_string, date_formats=None):
         TODO: Timezone issues
 
         """
+        global default_language_loader
+        if settings.SKIP_TOKENS != self._skip_tokens:
+            default_language_loader = LanguageDataLoader()
+            self = DateDataParser(*self._default_args)
```

Comment:
If `DateDataParser` instance's `_skip_tokens` differ from `settings.SKIP_TOKENS` it creates a new instance of `LanguageDataLoader` and assigns it to `default_language_loader` which is then used in `DataDataParser.__init__` while creating a new instance for `DateDataParser` with originally provided arguments and updating reference to self. This ensures new skip tokens are accounted for while detecting language and subsequently parsing date string.